### DISCUSSION TITLE
Type annotations

### DISCUSSION THREAD

**Discussion by Rudxain** [uBlockOrigin/uBlock-issues#3267] [Ideas]
Correct me if I'm wrong, but AFAIK (I've only read some source files), none of the `*.js` files have type annotations, let alone type-checking (such as `//@ts-check` or `jsconfig.json` from TypeScript). I find this quite surprising considering the quality and complexity of uBO source code.

Should I open an issue about this? I'm willing to help, by adding some JSDoc.

BTW, why are there 2 linters (`jshint` and `eslint`)? Why not simply use 1 with more lints enabled?

**Comment by ghost** [comment]
https://github.com/gorhill/uBlock/commit/36f81be13b6f1b2186b2da4080d7f1f16eaf7217

**Reply by Rudxain** [reply]
oh ok, so the migration is still in progress. There should be a tracking issue for that (I guess?)
